handler/tokenizer_handler.py

```diff
@@ -14,7 +16,15 @@
 def get_tokenizer_by_project(project_id: str) -> Language:
     project_item = project.get(project_id)
     tokenizer_config_str = project_item.tokenizer
-    return get_tokenizer(tokenizer_config_str)
+    tokenizer = get_tokenizer(tokenizer_config_str)
+
+    if get_config_value("is_managed"):
+        pickle_path = os.path.join("/inference", project_id, "tokenizer.pkl")
+        if not os.path.exists(pickle_path):
+            os.makedirs(os.path.dirname(pickle_path), exist_ok=True)
+            with open(pickle_path, "wb") as f:
+                pickle.dump(tokenizer, f)
```

Comment:
i would have assumed that this would be located in the init function not in the get. (Since it only needs to be checked and placed once not on every get)

Reply:
Yes, I see the problem. The tokenizer pickle files are not shared between the different gates project. So, a tokenizer pickle is needed in every inference/<project-id> directory. This is already on the list for improvements.
When changing this behavior, I recommend implementing the sharing of the same tokenizer in gates as well.
If we do not want to do that now, I could simply skip the tokenizer pickling if it already exists.